### PR TITLE
Softlayer Provider

### PR DESCRIPTION
**Pull request by rjeczalik**
Softlayer provider supports listing and modifying (tags) images.

Softlayer does not support removing images (at least the `SoftLayer_Virtual_Disk_Image` service) and it looks like each image is global in the Softlayer Object Storage, so no need for copying images (the image is built regionally by packer, but the archive is stored globally).

/cc @fatih

### CODE REVIEW

**Comment by fatih** [comment]
@rjeczalik this looks good. Please just add newlines after statements, otherwise its in a very dense and hard to follow.

**Comment by rjeczalik** [comment]
@fatih All addressed.

**Comment by fatih** [comment]
This is awesome :+1:  :)
